sdk/core/core-rest-pipeline/src/pipelineRequest.ts

```diff
@@ -74,6 +76,14 @@ export interface PipelineRequestOptions {
    */
   streamResponseStatusCodes?: Set<number>;
 
+  /**
+   * NODEJS ONLY
+   *
+   * A Node-only option to provide a custom `http.Agent`/`https.Agent`.
+   * Does nothing when running in the browser.
+   */
+  agent?: Agent;
```

Comment:
Do we have a specific reason why we don't have `agent` and `tlsSettings` in `PipelineRequestOptions`, or we missed adding them?

Reply:
Not _intentionally_ I would say, though it's not typically a thing we'd expect a consumer to have to set per operation as you generally would want to always use the same agent across all client calls so you can reuse the socket connection to the service.

Reply:
Our usual customers won't work at this level. I think writing custom pipeline policies is an advanced scenario and people need to understand what they are doing. I could add some warnings in the ref doc too.